src/renderer/component/threeViewer/index.jsx

```diff
@@ -0,0 +1,288 @@
+    this.wireframe = new THREE.LineSegments(wireframeGeometry, wireframeMaterial);
+    group.add(this.wireframe);
+  }
+
+  createMesh(geometry) {
+    const material = new THREE.MeshPhongMaterial({
+      opacity: 1,
+      transparent: true,
+      depthWrite: true,
+      vertexColors: THREE.FaceColors,
+      // Positive value pushes polygon further away
+      polygonOffsetFactor: 1,
+      polygonOffsetUnits: 1,
+    });
+
+    // Set material color
+    material.color.set(this.materialColors.green);
+
+    const mesh = new THREE.Mesh(geometry, material);
+
+    // Assign name
+    mesh.name = 'objectGroup';
+
+    this.scene.add(mesh);
+    this.fitMeshToCamera(mesh);
+    this.createWireFrame(mesh);
+    this.updateControlsTarget(mesh.position);
+    return mesh;
+  }
+
+  toggleWireFrame(show = false) {
+    this.wireframe.opacity = show ? 1 : 0;
+    this.mesh.material.opacity = show ? 0 : 1;
+  }
+
+  fitMeshToCamera(group) {
+    const max = { x: 0, y: 0, z: 0 };
+    const min = { x: 0, y: 0, z: 0 };
+
+    group.traverse(child => {
+      if (child instanceof THREE.Mesh) {
+        const box = new THREE.Box3().setFromObject(group);
+        // Max
+        max.x = box.max.x > max.x ? box.max.x : max.x;
+        max.y = box.max.y > max.y ? box.max.y : max.y;
+        max.z = box.max.z > max.z ? box.max.z : max.z;
+        // Min
+        min.x = box.min.x < min.x ? box.min.x : min.x;
+        min.y = box.min.y < min.y ? box.min.y : min.y;
+        min.z = box.min.z < min.z ? box.min.z : min.z;
+      }
+    });
+
+    const meshY = Math.abs(max.y - min.y);
+    const meshX = Math.abs(max.x - min.x);
+    const scaleFactor = 15 / Math.max(meshX, meshY);
+
+    group.scale.set(scaleFactor, scaleFactor, scaleFactor);
+    group.position.setY((meshY / 2) * scaleFactor);
+    group.position.multiplyScalar(-1);
+    group.position.setY((meshY * scaleFactor) / 2);
+  }
+
+  startLoader() {
+    const { source } = this.props;
+
+    if (source) {
+      ThreeLoader(source, this.renderModel.bind(this), {
+        onStart: this.handleStart(this),
+        onLoad: this.handleReady.bind(this),
+        onError: this.handleError.bind(this),
+        onProgress: this.handleProgress.bind(this),
+      });
+    }
+  }
+
+  handleStart() {
+    this.setState({ isLoading: true });
+  }
+
+  handleReady() {
+    // Handle load ready
+    this.setState({ isReady: true, isLoading: false });
+  }
+
+  handleResize = () => {
+    const { offsetWidth: width, offsetHeight: height } = this.viewer.current;
+    this.camera.aspect = width / height;
+    this.camera.updateProjectionMatrix();
+    this.controls.update();
+    this.renderer.setSize(width, height);
+  };
+
+  handleError() {
+    this.setState({ error: "Sorry, looks like we can't load this file" });
+  }
+
+  handleProgress() {
+    // const progress = (currentItem / totalItems) * 100;
+    // console.info(currentItem, totalItems, progress);
```

Comment:
I think we will hold off on the progress bar for now, we have a progress percentage below the file. 

I am going to add analytics for start to stream time, that will give us an idea of average time spent looking at the loading screen. If it's decently high I think we can add it.